### PR TITLE
[ISSUE 4] - Create Shopping List

### LINKED ISSUE #4: 4. As a user, I want to set up a new shopping list so I can start tracking purchased items.
## Summary 

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the token to `localStorage`
2. Save the list to our database using the `createList` function in `firebase.js` 
3. Show the user the list view
## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x]  When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Notes
- The `createList` function takes **three** arguments: the current user's id, the current user's email, and the name of the shopping list.
- The `setListPath` function takes **one** argument: the new list token.
- If you get stuck trying to redirect the user, check out the [useNavigate](https://reactrouter.com/en/6.14.2/hooks/use-navigate) hook from `react-rotuer`.

### PR DESCRIPTION
**Pull request by warrenchan13**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
This code allows users to create a list. When the 'Add List' button is clicked, an alert notifies the user whether the list has been successfully created or not.

## Related Issue
Closes #4  

## Acceptance Criteria
UI-related tasks:

 - [x] The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
  - [x] The input that accepts the name of the item has a semantic label element associated with it
  - [x] The user can submit this form with both the mouse and the Enter key
  - [x] When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

  - [x] Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.
  - [x] The shopping list path is stored in local storage using the setListPath function.
  - [x] Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes
`enhancement`

### Before
Screenshot 2024-08-22 at 2 00 47 PM

### After
Screenshot 2024-08-22 at 12 58 18 PM
Screenshot 2024-08-22 at 12 58 31 PM
Screenshot 2024-08-22 at 1 02 41 PM
Screenshot 2024-08-22 at 1 02 56 PM
Screenshot 2024-08-22 at 1 03 07 PM

## Testing Steps / QA Criteria
1. Sign In
2. Enter a new list and click submit
3. Check Firebase to see if it saved a token under userID
4. Check if the new list shows up on the home page
5. Test alert if user submits an empty list
6. Test alert if user submits a duplicate list

### CODE REVIEW

**Review by kweeuhree — APPROVED** [review]
Looks great to me, appreciate edge cases tests!

**Review by dilekbaykara — APPROVED** [review]
Looks great, the tooltip alert in the input field is a nice touch as well!

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,13 +1,59 @@
...
+		if (currentLists.includes(listName.toLowerCase())) {
+			alert('The list already exists. Please enter a different name.');
+			setListName('');
+			return;
+		}
+
+		try {
+			const response = await createList(userId, userEmail, listName);
+			// console.log('check response', response);
+			setListPath(response);
```
Comment (jendevelops) [inline src/views/Home.jsx:30]: Nit: if the `listPath` is returned from `createList`, I would name the variable `listPath` to make it more clear
```suggestion
			const listPath = await createList(userId, userEmail, listName);
			// console.log('check response', response);
			setListPath(listPath);
```

File: src/views/Home.jsx [hunk 2]
```diff
@@ -1,13 +1,59 @@
...
+
+		if (currentLists.includes(listName.toLowerCase())) {
+			alert('The list already exists. Please enter a different name.');
+			setListName('');
+			return;
+		}
+
+		try {
+			const response = await createList(userId, userEmail, listName);
+			// console.log('check response', response);
```
Comment (jendevelops) [inline src/views/Home.jsx:29]: Hanging console log, please delete

**Review by jendevelops — APPROVED** [review]
Great job with this PR! I tested out the functionality and it works beautifully.

I had a suggestion about variable naming. Particularly "naming with a purpose". It's very easy to name a promise response as `response`, but naming variables is actually one of the tools outside of commenting your code that can help convey context/ intent to your other team members. Whenever in doubt, you can always ask during code review as well if a number of options makes more sense as you grow this muscle.
